File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class SmartphoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")
        self.subscriber_ = self.create_subscription(String, "robot_news", self.callback_read_message, 10)
        self.get_logger().info("Beep Bop, smartphone is up!")

    def callback_read_message(self, msg: String):
        self.get_logger().info("[Recieved message]: " + msg.data)

def main(args=None):
    rclpy.init(args=args)

    node = SmartphoneNode()

    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/smartphone.cpp

#include <format>

#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::placeholders;

class SmartphoneNode : public rclcpp::Node 
{
public:
    SmartphoneNode() : Node("smartphone")
    {
        subscriber = this->create_subscription<example_interfaces::msg::String>(
            "robot_news", 
            10,
            [this](const example_interfaces::msg::String::SharedPtr msg) { 
                this->robotNewsListener(msg); 
            }
        );

        RCLCPP_INFO(this->get_logger(), "Smartphone is listening");

    }
private:
    void robotNewsListener(const example_interfaces::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), std::format("{}", msg->data).c_str());
    }
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber;
};

int main(int argc, char ** argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<SmartphoneNode>();

    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}